src/ctia/entity/incident/es_store.clj

```diff
@@ -0,0 +1,106 @@
+(ns ctia.entity.incident.es-store
+  (:require [ctia.entity.incident.schemas
+             :refer [PartialStoredIncident StoredIncident]]
+            [ctia.store :refer [IQueryStringSearchableStore IStore]]
+            [ctia.stores.es.mapping :as em]
+            [ctia.stores.es.store :refer [close-connections!]]
+            [ctia.stores.es.crud :as crud]
+            [clojure.string :as string]
+            [schema.core :as s]
+            [ctia.schemas.search-agg :refer [SearchQuery]]))
+
+(def incident-mapping
+  {"incident"
+   {:dynamic false
+    :properties
+    (merge
+     em/base-entity-mapping
+     em/describable-entity-mapping
+     em/sourcable-entity-mapping
+     em/stored-entity-mapping
+     {:confidence em/token
+      :status em/token
+      :incident_time em/incident-time
+      :categories em/token
+      :discovery_method em/token
+      :intended_effect em/token
+      :assignees em/token
+      :promotion_method em/token})}})
+
+(def handle-create (crud/handle-create :incident StoredIncident))
+(def handle-update (crud/handle-update :incident StoredIncident))
+(def handle-read (crud/handle-read PartialStoredIncident))
+(def handle-delete (crud/handle-delete :incident))
+(def handle-list (crud/handle-find PartialStoredIncident))
+(def handle-bulk-delete crud/bulk-delete)
+(def handle-query-string-search (crud/handle-query-string-search PartialStoredIncident))
+(def handle-query-string-count crud/handle-query-string-count)
+(def handle-aggregate crud/handle-aggregate)
+(def handle-delete-search crud/handle-delete-search)
+
+(s/defn append-query-clauses :- SearchQuery
+  [search-query  :- SearchQuery
+   str-lucene-clauses :- [s/Str]]
+  (let [query (get-in search-query [:full-text :query])
+        prepared (string/join " OR " str-lucene-clauses)
+        lucene-query (if query
+                       ;; from previous observations the order of filters in lucene matters
+                       ;; prepared is likely more optimized, add it first
+                       (format "(%s) AND (%s)" prepared query)
+                       prepared)]
+    (assoc-in search-query [:full-text :query] lucene-query)))
+
+(defn high-impact-sources
+  [conn-state]
+  (let [get-in-config (get-in conn-state [:services :ConfigService :get-in-config])]
+    (some-> (get-in-config [:ctia :incident :high-impact :source])
+            (string/split #","))))
+
+(s/defn prepare-impact :- SearchQuery
```

Comment:
While I totally understand what this is doing and the why, I think we should be safer to add a new query parameter instead of adding a trick in the filter-map, because this is a non-trivial, surprise-specific behavior.

Why not consider:

adding a new query parameter in the search routes for incidents, a boolean called `high-impact`. This way you wouldn't pollute the behavior of `filter-map`.

From a client standpoint, this looks surprising to me.

The difficult part for me is the `(update :filter-map dissoc :high_impact)`.